packages/provider/wrapper.js

```diff
@@ -32,6 +32,43 @@ module.exports = {
       const originalSendAsync = provider.sendAsync.bind(provider);
       provider.sendAsync = this.send(originalSendAsync, preHook, postHook);
     }
+
+    // EIP-1193 provider wrapping
+    if (
+      typeof provider.request === "undefined" &&
```

Comment:
I don't believe you need to do the `typeof` thing here. I think you can just do `provider.request === undefined`.